### PR TITLE
Version 1.3

### PR DESCRIPTION
**Pull request by micahflee**
This includes all commits between version 1.2 and 1.3.

### CODE REVIEW

File: install/get-tor-osx.py [hunk 1]
```diff
@@ -28,9 +28,9 @@
 import urllib.request
 
 def main():
-    dmg_url = 'https://archive.torproject.org/tor-package-archive/torbrowser/7.0.11/TorBrowser-7.0.11-osx64_en-US.dmg'
-    dmg_filename = 'TorBrowser-7.0.11-osx64_en-US.dmg'
-    expected_dmg_sha256 = '5143e4a2141a69f66869be13eef4bcaac4e6c27c78383fc8a4c38b334759f3a2'
+    dmg_url = 'https://archive.torproject.org/tor-package-archive/torbrowser/7.5/TorBrowser-7.5-osx64_en-US.dmg'
+    dmg_filename = 'TorBrowser-7.5-osx64_en-US.dmg'
+    expected_dmg_sha256 = '43a8dc0afd0a77e42766311eb54ad9fc8714f67fcd2d3582a3bcb98b22c2e629'
```
Comment (micahflee) [inline install/get-tor-osx.py]: Re-downloaded dmg, checked signature, confirmed the sha256 matches

File: install/get-tor-windows.py [hunk 2]
```diff
@@ -28,9 +28,9 @@
 import urllib.request
 
 def main():
-    exe_url = 'https://archive.torproject.org/tor-package-archive/torbrowser/7.0.11/torbrowser-install-7.0.11_en-US.exe'
-    exe_filename = 'torbrowser-install-7.0.11_en-US.exe'
-    expected_exe_sha256 = 'a033eb9b9ed2ad389169b36a90946a8af8f05bd0c7bbd3e37678041331096624'
+    exe_url = 'https://archive.torproject.org/tor-package-archive/torbrowser/7.5/torbrowser-install-7.5_en-US.exe'
+    exe_filename = 'torbrowser-install-7.5_en-US.exe'
+    expected_exe_sha256 = '81ccb9456118cf8fa755a3eafb5c514665fc69599cdd41e9eb36baa335ebe233'
```
Comment (micahflee) [inline install/get-tor-windows.py]: Re-downloaded exe, checked signature, confirmed the sha256 matches

File: onionshare_gui/update_checker.py [hunk 3]
```diff
@@ -96,7 +99,10 @@ def check(self, force=False, config=False):
                 if force:
                     path += '?force=1'
 
-                onion_domain = 'elx57ue5uyfplgva.onion'
+                if Version(self.onion.tor_version) >= Version('0.3.2.9'):
+                    onion_domain = 'lldan5gahapx5k7iafb3s4ikijc4ni7gx5iywdflkba5y2ezyg6sjgyd.onion'
+                else:
+                    onion_domain = 'elx57ue5uyfplgva.onion'
```
Comment (micahflee) [inline onionshare_gui/update_checker.py]: Confirmed these are the correct .onions

**Comment by micahflee** [comment]
I realized that because I created this PR I can't actually "review" it in github myself. But I did just manually review it and so far (we still have at least one more PR to merge in) everything looks good.

All of this should have already been reviewed, but it's prudent to check again. I basically just skimmed the diff to make sure nothing looks obviously suspicious or malicious, and that all code being modified is code that we actually should have touched to fix bugs and add features this release. I also re-verified the Tor Browser sha256 checksums, and made sure the onionshare.org .onions are correct.

**Comment by mig5** [comment]
Let me know how your experience is on Windows in terms of sharing in various scenarios? 

Twice now today, I have found onionshare-gui.exe go unresponsive after trying to start a share. But it could totally be my VM, I'm not sure there's a real issue. Just curious to see how it is for others.

**Comment by micahflee** [comment]
Here's a list of stuff to do for the release:

- Merge #644 in
- Test sharing and downloading in Linux, Windows, and macOS
- Build the installer in Windows, and test to make sure it installs everything properly, the installed version actually works and all resources are included, and the uninstall deletes all files properly
- Build the macOS installer and make sure it works to install, and the installed version works

Once that testing is done:

- Merge `develop` into `master`
- Make a signed tag for the release
- Build a codesigned macOS package, and generate a PGP signature
- Build a codesigned Windows package, and generate a PGP signature
- Create a github release that includes the changelog, upload the Windows and Mac binaries and the sigs
- Make a PPA release for Linux, and alert Fedora and Debian maintainers there's a new release
- Update onionshare-website to include links to the new release

Any steps I'm missing?

**Comment by mig5** [comment]
* Add the PGP signatures as direct downloads on the Onionshare.org site so they can be fetched via the v3 onion service (probably a good idea)
 * Tweet and announce on mailing list


If it helps you, here is my list of 'tests' I performed manually yesterday or the day before (obviously not perfect, since I missed bugs that have been fixed since!):

1. Test regular share (and that the web server/onion service has definitely stopped once the share stops, that all functionality in the UI returns for the user e.g adding/deleting files, accessing Settings button)

2. Test share with 'stay open' mode (and test simultaneous downloads of a file at least 5MB all succeed)

3. Test share with persistent URL (second share shows same URL)

4. Test share with stealth mode (hidservauth link appears, service can't be reached without it, etc)

5. Test share with shutdown timer,
5a. Test shutdown timer with stay open mode (it should still shutdown when the timer runs out despite stay open mode)
5b. Test shutdown timer with large file (it should wait for download to finish)
5c. Test shutdown timer fails to start if the timeout has passed already by the time we click start
5d. Test shutdown timer fails to start if the timeout passes whilst it was still preparing to start

6. Test bridges

7. Test adding 4 files, deleting 1 file in the middle, and that a subsequent share of those files shows that same middle file is not present (not that we deleted the 'wrong' file)

8. Test that adding a non-readable file (e.g chown root:root and chmod 400) cannot be added

9. Test killing Tor when connected, and that various 'Disconnected from Tor' stuff appears

10. Test drag-n-drop

11. Settings Dialog: test that the Help button and 'More Info' (Stealth) links actually open in browser

12. Settings Dialog: test the 'Test Tor Settings' widget works

13. Settings Dialog: test the Update Checker (Mac/Windows)

A lot of these are covered by my GUI unit tests, you are welcome to run the job at https://jenkins.mig5.net:8443/view/OnionShare/job/Test_onionshare_ux-update_branch/ (ignore the name, it now tracks develop branch) - it sometimes fails on typical tor stuff like socket timeout trying to build a circuit to the onion service to download the share

One I ran just now: https://jenkins.mig5.net:8443/view/OnionShare/job/Test_onionshare_ux-update_branch/17/console

**Comment by Baccount** [comment]
`Add the PGP signatures as direct downloads on the Onionshare.org site so they can be fetched via the v3 onion service (probably a good idea). `
Is #555 related to your statement above?

**Comment by mig5** [comment]
Yes I had that in mind, as well as #523 which we may be implementing soon

**Pull request merged**